### ISSUE TITLE
Scipy Optimizer trust-constr does not work

### ISSUE THREAD

**Issue by LeonardSchmiester** [ICB-DCM/pyPESTO#613]
Optimization with `trust-constr` from scipy does not work currently because the default options we set for scipy are not known to `trust-constr`. Therefore optimization stops with:
```
start 0 failed: _minimize_trustregion_constr() got an unexpected keyword argument 'max_nfev'
```

@yannikschaelte pointed out, that also for some other optimizer there are warnings regarding not supported options (see https://github.com/ICB-DCM/pyPESTO/runs/2225587545?check_suite_focus=true#step:6:1169)

**Comment by yannikschaelte** [comment]
I think we'll want a more intelligent method-specific setting of hyperparameters.

**Comment by yannikschaelte** [comment]
@LoosC , you here?

**Comment by FFroehlich** [comment]
This is probably the result of scipy better harmonizing the optimization interfaces. Simply removing the specific mapping of `maxiter` to `max_nfev` for trust_constr should fix the issue.

**Comment by FFroehlich** [comment]
`maxiter`/`max_nfev` appears to be fixed in https://github.com/ICB-DCM/pyPESTO/pull/283 (about a year before this issue was posted?). I think the actual issue is `ftol`, which shouldn't be set with a default for `trust-constr`.

**Comment by yannikschaelte** [comment]
@PaulJonasJost this should be related to #778

**Comment by PaulJonasJost** [comment]
> @PaulJonasJost this should be related to #778

@yannikschaelte, @LeonardSchmiester could this perhaps have been fixed by #803?
